src/posix/threading.c

```diff
@@ -173,12 +176,15 @@ static void *_thread_init(void *arg)
 }
 
 
-sys_thread_t *sys_thread(void *(*run_function) (void *), void *arg)
+sys_thread_t *sys_thread(const char * thread_name, void *(*run_function) (void *), void *arg)
 {
+    // thread name may be at most 16 bytes, counting the final \0 byte
+    assert(strlen(thread_name) < 16);
     sys_thread_t *thread = NEW(sys_thread_t);
     thread->f = run_function;
     thread->arg = arg;
     pthread_create(&(thread->thread), 0, _thread_init, (void*) thread);
+    pthread_setname_np(thread->thread, thread_name);
```

Comment:
return code from these should be checked

Reply:
Not a bad idea to check the return codes, but I wouldn't hold up this patch for it.

Usually for those functions that should always work unless we pass garbage to them then doing a check only in the debug build case would catch any misuse during development.  You'll see code that does this alot:

int result = foo(x);
(void)result; assert(result == 0);